patches/server/0966-Expand-PlayerDeathEvent-API.patch

```diff
@@ -0,0 +1,62 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: dawon <dawon@dawon.eu>
+Date: Sun, 12 Mar 2023 17:17:40 +0100
+Subject: [PATCH] Expand PlayerDeathEvent API
+
+
+diff --git a/src/main/java/net/minecraft/server/level/ServerPlayer.java b/src/main/java/net/minecraft/server/level/ServerPlayer.java
+index 289429eb464548acc80262a49444f49f8f57fc0c..83f81e3bdb7ab077095dd18059f4be4f30357b77 100644
+--- a/src/main/java/net/minecraft/server/level/ServerPlayer.java
++++ b/src/main/java/net/minecraft/server/level/ServerPlayer.java
+@@ -905,7 +905,7 @@ public class ServerPlayer extends Player {
+ 
+         String deathmessage = defaultMessage.getString();
+         this.keepLevel = keepInventory; // SPIGOT-2222: pre-set keepLevel
+-        org.bukkit.event.entity.PlayerDeathEvent event = CraftEventFactory.callPlayerDeathEvent(this, loot, PaperAdventure.asAdventure(defaultMessage), defaultMessage.getString(), keepInventory); // Paper - Adventure
++        org.bukkit.event.entity.PlayerDeathEvent event = CraftEventFactory.callPlayerDeathEvent(this, loot, PaperAdventure.asAdventure(defaultMessage), defaultMessage.getString(), keepInventory, flag); // Paper - Adventure
+         // Paper start - cancellable death event
+         if (event.isCancelled()) {
+             // make compatible with plugins that might have already set the health in an event listener
+@@ -922,13 +922,14 @@ public class ServerPlayer extends Player {
+         }
+ 
+         net.kyori.adventure.text.Component deathMessage = event.deathMessage() != null ? event.deathMessage() : net.kyori.adventure.text.Component.empty(); // Paper - Adventure
++        Component deathScreenMessage = PaperAdventure.asVanilla(event.deathScreenMessage() != null && event.shouldShowDeathMessage() ? event.deathScreenMessage() : net.kyori.adventure.text.Component.empty()); // Paper - death screen api
```

Comment:
This can just be a local var above the `ClientboundPlayerCombatKillPacket` constructor. It doesn't have to exist outside the scope of the following if statement. Just a simple
```java
event.deathScreenMessage() != null ? PaperAdventure.asVanilla(event.deathScreenMessage()) : ichatbasecomponent
```